sos/report/plugins/kdump.py

```diff
@@ -124,4 +124,49 @@ def setup(self):
         if self.get_option("collect-kdumps"):
             self.add_copy_spec(["/var/kdump-*"])
 
+
+class AzureKDump(KDump, AzurePlugin):
+
+    files = ('/etc/kdump.conf',)
+    packages = ('kexec-tools',)
+
+    option_list = [
+        PluginOpt("get_vm_core", default=False, val_type=bool,
+                  desc="collect vm core")
+    ]
+
+    def read_kdump_conffile(self):
+        """ Parse /etc/kdump file """
+        path = "/var/crash"
+
+        kdump = '/etc/kdump.conf'
+        with open(kdump, 'r', encoding='UTF-8') as file:
+            for line in file:
+                if line.startswith("path"):
+                    path = line.split()[1]
+
+        return path
+
+    def setup(self):
+        super().setup()
+
+        self.add_copy_spec([
+            "/etc/kdump.conf",
+            "/usr/lib/udev/rules.d/*kexec.rules"
+        ])
+
+        try:
+            path = self.read_kdump_conffile()
+        except Exception:  # pylint: disable=broad-except
+            # set no filesystem and default path
+            path = "/var/crash"
+
+        self.add_cmd_output(f"ls -alhR {path}")
+        self.add_copy_spec(f"{path}/*/vmcore-dmesg.txt")
+        self.add_copy_spec(f"{path}/*/kexec-dmesg.log")
+
+        # collect the latest vmcore created in the last 24hrs <= 2GB
+        if self.get_option("get_vm_core"):
+            self.add_copy_spec(f"{path}/*/vmcore", sizelimit=2048, maxage=24)
```

Comment:
The use of `sizelimit` here may be undesirable. With `add_copy_spec()`, this parameter tells sos to copy the last X MB of the file if it is over the given size, which would result in an unusable binary blob in the archive taking up 2GB of space.

What we would want here is to either capture a usable vmcore, or nothing at all, so rather than the use of `sizelimit`, we should get a list of potential vmcore(s) to copy, check their size and age (e.g. `os.stat()`), and only then pass the path to an `add_copy_spec()` call.

`maxage` makes sense here.

Reply:
This is a great point. When I was testing actually, I saw the "File is over size limit and is bianry" log - So, I think the last X MB is not copied?

https://github.com/sosreport/sos/blob/main/sos/report/plugins/__init__.py#L1854C1-L1860C37

Please correct me if I am wrong. Based on this, I think the whole binary is copied or nothing at all is copied :)

Reply:
Oh, duh. We did change that default a while back. I swear some days I don't know where my head is at.

Reply:
No worries at all :) 

Glad we're on the same page!